cmd/server.go

```diff
@@ -103,31 +94,90 @@ func serverOptsFor(cmd *cobra.Command) (map[string]plugin.Root, server.Opts, err
 		return nil, server.Opts{}, err
 	}
 
-	// Unmarshal the external plugins, if any are specified
-	var externalPlugins []plugin.ExternalPluginSpec
-	if err := viper.UnmarshalKey("external-plugins", &externalPlugins); err != nil {
-		return nil, server.Opts{}, fmt.Errorf("failed to unmarshal the external-plugins key: %v", err)
-	}
-
-	// Load internal plugins that are not specifically excluded.
-	enabledPlugins := viper.GetStringSlice("plugins")
 	plugins := make(map[string]plugin.Root)
-	if len(enabledPlugins) > 0 {
+
+	// Check the internal plugins
+	if viper.IsSet("plugins") || viper.IsSet("external-plugins") {
```

Comment:
External plugins were added here b/c it is possible (and makes sense) for someone to load only external plugins